src/commands/deployment/gate.ts

```diff
@@ -207,47 +211,61 @@ export class DeploymentGateCommand extends Command {
     return request
   }
 
-  private async requestGateEvaluation(api: APIHelper, request: GateEvaluationRequest): Promise<string> {
+  private async requestGateEvaluation(
+    api: APIHelper,
+    request: GateEvaluationRequest,
+    timeout: number
+  ): Promise<string> {
     this.logger.info('Requesting gate evaluation...')
 
-    try {
-      const response = await api.requestGateEvaluation(request)
-      const evaluationId = response.data.data.attributes.evaluation_id
-      this.logger.info(chalk.green(`Gate evaluation started successfully. Evaluation ID: ${evaluationId}\n`))
+    const doRequest = async () => {
+      try {
+        const response = await api.requestGateEvaluation(request)
+        const id = response.data.data.attributes.evaluation_id
+        this.logger.info(chalk.green(`Gate evaluation started successfully. Evaluation ID: ${id}\n`))
+
+        return id
+      } catch (error) {
+        if (isAxiosError(error) && error.response?.status) {
+          this.logger.error(`Request failed with error: ${error.response.status} ${error.response.statusText}`)
+        } else {
+          this.logger.error(
+            `Could not start gate evaluation with unknown error: ${
+              error instanceof Error ? error.message : String(error)
+            }`
+          )
+        }
 
-      return evaluationId
-    } catch (error) {
-      if (isAxiosError(error) && error.response?.status) {
-        this.logger.error(`Request failed with client error: ${error.response.status} ${error.response.statusText}`)
-      } else {
-        this.logger.error(
-          `Could not start gate evaluation with unknown error: ${
-            error instanceof Error ? error.message : String(error)
-          }`
-        )
+        throw error
       }
-
-      throw error
     }
+
+    const evaluationId = await retryRequest(doRequest, {
+      ...this.getRetryOptions(timeout),
+      onRetry: (e, attempt) => {
+        this.logger.info(`Retrying gate evaluation request (${attempt} attempts)...`)
+      },
+    })
+
+    return evaluationId
   }
 
   private async pollForEvaluationResults(
     api: APIHelper,
     evaluationId: string,
-    timeoutSeconds: number
+    timeout: number
```

Comment:
So the overall behavior here is:
- We poll the endpoint every 15 seconds
- Each time we call the endpoint, we might do N calls if the endpoint is unavailable? 

Could we just wait the next call and threat it as a 404/in-progress perhaps?